lib/streamlit/elements/skeleton.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, cast
+
+from streamlit.proto.Skeleton_pb2 import Skeleton as SkeletonProto
+from streamlit.runtime.metrics_util import gather_metrics
+
+if TYPE_CHECKING:
+    from streamlit.delta_generator import DeltaGenerator
+
+# Works just like an `empty` except animated by default.
+class SkeletonMixin:
+    @gather_metrics("skeleton")
+    def skeleton(self, height: int = 107) -> "DeltaGenerator":
```

Comment:
Hmm, the seemingly arbitrary value of 107 here makes me feel a bit uneasy (I guess it's not much more arbitrary than using `"9rem"` as we currently do, but that feels less bad to me for whatever reason -- likely because hard-coded CSS values are relatively common in those contexts).

I wonder if there's some way to make this feel like less of a magic number, although I'm not sure myself what the best way of addressing this would be

Reply:
Turns out, 107 _is_ `"9rem"` in CSS points given streamlit's default font sizes. "pt" is also what all the other streamlit protos use as their dimensional units (at least based on what I saw). That's why this is 107 by default, so it looks just like the original.

Some other options could be:
* Make height a string and add some kind of validator that ensures the string is a valid CSS dimension
* Make height REMs again, but allow floats

Reply:
yeah, 107 is a bit strange, but I think it is the best way to go. I believe there is also no specific reason to use `9rem` other than that it looked good. We probably can also just use 105 or 100 here.